var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -51,7 +51,7 @@ def configure_args(self):
 
         # http://lists.gnu.org/archive/html/bug-m4/2016-09/msg00002.html
         arch = spec.architecture
-        if (arch.platform == 'darwin' and arch.platform_os == 'sierra' and
+        if (arch.platform == 'darwin' and arch.os == 'sierra' and
```

Comment:
@gzagaris This looks like it will allow the package to build (avoiding an attribute error for `platform_os`) but on my Mac I see the `os` as `highsierra`. That is, if I do for example `spack spec bzip2` I see `darwin-highsierra-x86_64` as my architecture. Are you seeing something different? If you are seeing the same thing as me, I think the check should be `arch.os == 'highsierra'`.

Reply:
@scheibelp, the value of `arch.os` depends on the version of macOS that you are running:

- For macOS 10.12, `arch.os == sierra`
- For macOS 10.13, `arch.os == highsierra`

This logic was specifically applied to fix an issue for macOS 10.12, i.e., `arch.os == sierra`.  See  de7be66.

I am running macOS 10.13, and it works fine for me without applying this specialized logic.